airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -46,37 +50,62 @@ class ManifestDeclarativeSource(DeclarativeSource):
 
     VALID_TOP_LEVEL_FIELDS = {"check", "definitions", "schemas", "spec", "streams", "type", "version"}
 
-    def __init__(self, source_config: ConnectionDefinition, debug: bool = False):
+    def __init__(self, source_config: ConnectionDefinition, debug: bool = False, construct_using_pydantic_models: bool = False):
         """
         :param source_config(Mapping[str, Any]): The manifest of low-code components that describe the source connector
         :param debug(bool): True if debug mode is enabled
         """
         self.logger = logging.getLogger(f"airbyte.{self.name}")
 
+        # Controls whether we build components using the manual handwritten schema and Pydantic models or the legacy flow
+        self.construct_using_pydantic_models = construct_using_pydantic_models
+
+        # For ease of use we don't require the type to be specified at the top level manifest, but it should be included during processing
+        manifest = dict(source_config)
+        if "type" not in manifest:
+            manifest["type"] = "DeclarativeSource"
+
         evaluated_manifest = {}
-        resolved_source_config = ManifestReferenceResolver().preprocess_manifest(source_config, evaluated_manifest, "")
-        self._source_config = resolved_source_config
+        resolved_source_config = ManifestReferenceResolver().preprocess_manifest(manifest, evaluated_manifest, "")
+        propagated_source_config = ManifestComponentTransformer().propagate_types_and_options("", resolved_source_config, {})
+        self._new_source_config = propagated_source_config
+        self._legacy_source_config = resolved_source_config
```

Comment:
In the immediate, it's a little ugly to retain both the legacy manifest and the new manifest (the key difference being the new manifest goes through the transformer to propagate options and assign types). But we will be getting rid of `_legacy_source_config` in the final part of the refactor anyway. Doing it this way just helps make it very explicit what lines of code will get deleted in that PR.

The final flow after we finish will look like this:
```
evaluated_manifest = {}
resolved_source_config = ManifestReferenceResolver().preprocess_manifest(manifest, evaluated_manifest, "")
propagated_source_config = ManifestComponentTransformer().propagate_types_and_options("", resolved_source_config, {})
self._source_config = propagated_source_config
self._debug = debug
self._constructor = ModelToComponentFactory() # New factory which converts the manifest to Pydantic models to construct components
        
self._validate_source()
```